aspnetcore/tutorials/razor-pages/search.md

```diff
@@ -0,0 +1,113 @@
+
+Note: The [Contains](http://msdn.microsoft.com/library/bb155125.aspx) method is run on the database, not in the c# code. The case sensitivity on the query depends on the database and the collation. On SQL Server, `Contains` maps to [SQL LIKE](https://docs.microsoft.com/en-us/sql/t-sql/language-elements/like-transact-sql), which is case insensitive. In SQLlite, with the default collation, it's case sensitive.
+
+Navigate to the Movies page and append a query string such as `?searchString=Ghost` to the URL (for example, `http://localhost:5000/Movies?searchString=Ghost`). The filtered movies are displayed.
+
+![Index view](search/_static/ghost.png)
+
+If you add the following route template to the Index page, the search string can be passed as a URL segment (for example, `http://localhost:5000/Movies/ghost`).
+
+```cshtml
+@page "{searchString?}"
+```
+
+The preceding  route constraint allows searching the title as route data (a URL segment) instead of as a query string value.
+
+![Index view with the word ghost added to the Url and a returned movie list of two movies, Ghostbusters and Ghostbusters 2](search/_static/g2.png)
+
+However, you can't expect users to modify the URL to search for a movie. In this step you add UI to filter movies. If you added the route constraint `"{searchString?}"`, remove it.
+
+Open the *Pages/Movies/Index.cshtml* file, and add the `<form>` markup highlighted below:
+
+[!code-cshtml[Main](razor-pages-start/sample/RazorPagesMovie/Pages/Movies/Index2.cshtml?highlight=14-19&range=1-22)]
+
+The HTML `<form>` tag uses the [Form Tag Helper](xref:mvc/views/working-with-forms#the-form-tag-helper). When the form is submitted, the filter string is sent to the Pages/Movies/Index Razor Page. Save the changes and test the filter.
+
+![Index view with the word ghost typed into the Title filter textbox](search/_static/filter.png)
+
+## Search by genre
+
+Add the the following highlighted properties to the *Pages/Movies/Index.cshtml.cs* file:
+
+[!code-csharp[Main](razor-pages-start/sample/RazorPagesMovie/Pages/Movies/Index.cshtml.cs?name=snippet_newProps&highlight=11-)]
+
+The `SelectList Genres` containing the list of genres. This allows the user to select a genre from the list.
```

Comment:
I'm not sure what the following sentence is intended to say. Is there a word missing?

"The `SelectList Genres` containing the list of genres."